<a href="https://colab.research.google.com/github/Olmedann/Curso-de-python-para-matematicas-2023/blob/main/Numeros_de_Fibonacci_(conjeturas_sobre_potencias).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Para escribir una función que tenga un argumento obligatorio n y un argumento opcional pot, y que regrese los números de fibonacci menores que n que sean una potencia pot exacta hacemos los siguiente:

In [1]:
def es_pot(n,k):
    return int((n)**(1/k)) ** k == n

def potencias_fibonacci(n,pot=2):
    fibs = []
    a, b = 0, 1
    while a < n:
        if es_pot(a,pot):
            fibs.append(a)
        a, b = b, a+b
    return fibs

potencias_fibonacci(10**20,2)

[0, 1, 1, 144]

Observe que si corremos el código anterior obtenemos que los números de Fibonacci que son cuadrados menores que $10^{20}$ son 0,1 y 144, mientras que los cubos que encontramos menores que  $10^{20}$ son 0,1,8. Veamos que si consideramos un número mayor, seguimos obteniendo los mismos números, lo cual permite formular la siguiente conjetura.

**Conjetura**: Los únicos números de Fibonacci que son cuadrados son 0,1 y 144, los únicos cubos son 0,1,8 y los únicos cuartos son 0,1.
Dicha conjetura ya se ha demostrado y puede encontrar más información al respecto en el siguiente artículo: https://www.sciencedirect.com/science/article/abs/pii/B9780080119908500095$10^{20}$
